src/collection-view.js

```diff
@@ -101,7 +101,7 @@ Marionette.CollectionView = Marionette.View.extend({
   _onCollectionAdd: function(child, collection, opts) {
     var index;
     if (opts.at !== undefined) {
-      index = opts.at;
+      index = opts.index !== undefined ? opts.index : collection.indexOf(child);
```

Comment:
This'll fail for filtered collections. Falling into the `else` path below should work.

```js
 if (opts.index !== undefined) {
   index = opts.index;
 } else {
   index = _.indexOf(this._filteredSortedModels(), child);
 }
```